mono/metadata/Makefile.am

```diff
@@ -181,9 +177,40 @@ libmono_system_native_la_SOURCES = \
 	../../external/corefx/src/Native/Unix/System.Native/pal_tcpstate.c \
 	../../external/corefx/src/Native/Unix/System.Native/pal_tcpstate.h \
 	../../external/corefx/src/Native/Unix/System.Native/pal_random.c \
-	../../external/corefx/src/Native/Unix/System.Native/pal_random.h \
-	pal-icalls.h \
-	pal-icalls.c
+	../../external/corefx/src/Native/Unix/System.Native/pal_random.h
+
+system_native_ios_sources = \
+	pal-ios.h \
+	pal-ios.c
+
+system_native_android_sources = \
+	pal-android.h \
+	pal-android.c
+
+libmono_system_native_la_SOURCES = $(system_native_common_sources)
```

Comment:
It used to be problematic - a very long time ago - to conditionalize the `_SOURCES` in this way as a `make dist` would fail to pickup all the files.  Changes are that modern-era `automake` will handle this correctly, so if this builds then we don't have to worry about it.

If you run into any problems, then you can conditionally define `system_native_sources` and use `libmono_system_native_la_SOURCES = $(system_native_sources)`